packages/bundle/src/core/location.ts

```diff
@@ -118,9 +118,15 @@ export const redirectToPage = async (redirect, meta) => {
     suggestion: meta.get('q'),
   });
 
+  if (redirect.get('url').startsWith('https')) {
+    document.location.href = redirect.get('url');
+  }
+
+  const origin = document.location.origin + __root.config.getIn(['location', 'defaultPath'], '');
   if (isHistoryChanged()) {
     return getHistory().push(
-      redirect.get('url').replace(document.location.origin, ''),
+      redirect.get('url')
+        .replace(origin, ''),
       { type: 'FindifyUpdate' }
     );
   }
```

Comment:
@wolff95 mb if history didn't changed we also need to clean origin from redirect url ?

Reply:
@timaramazanov @wolff95 if SA redirect.url returns the full url always, then we should always clean origin imo, but not sure as I did not run a full test

Reply:
@timaramazanov I don't know if I understood you here.
As in, if history changed and so, if a user submitted a query with such redirect, it would be redirect, in the case it is not changed but still redirectToPage has been called we simply overwrite the href directly with what comes in. TBH, I don't know how it would work to redirect without having a historyChange